dev/com.ibm.ws.session.cache_fat/test-applications/sessionCacheApp/src/session/cache/web/SessionCacheTestServlet.java

```diff
@@ -314,7 +400,96 @@ public void testSerialization_complete(HttpServletRequest request, HttpServletRe
         }
         String attributeNamesString = attributeNames.toString();
         assertTrue(attributeNamesString, attributeNames.containsAll(Arrays.asList("map", "str", "appObject")));
-        assertEquals(attributeNamesString, weldCount + 3, attributeNames.size());
+        assertEquals(attributeNamesString, weldCount + 27, attributeNames.size());
+
+        //Test the serialization of performance improved primitives (that aren't tested elsewhere)
+        java.util.Date actualDate = (Date) session.getAttribute("date");
+        Calendar cal = Calendar.getInstance();
+        cal.clear();
+        cal.set(2018, 4, 19, 9, 23, 45);
+        java.util.Date expectedDate = cal.getTime();
+        assertEquals(expectedDate.getTime(), actualDate.getTime());
+
+        java.sql.Date expectedSqlDate = new java.sql.Date(1524236907331l);
+        java.sql.Date actualSqlDate = (java.sql.Date) session.getAttribute("sqlDate");
+        assertEquals(expectedSqlDate.getTime(), actualSqlDate.getTime());
+
+        double actualDouble = (double) session.getAttribute("double");
+        assertEquals(55901.55902, actualDouble, 0);
+
+        float actualFloat = (float) session.getAttribute("float");
+        assertEquals(12345.6789f, actualFloat, 0);
+
+        byte expectedByte = Byte.parseByte("3");
+        byte actualByte = (byte) session.getAttribute("byte");
+        assertEquals(expectedByte, actualByte);
+
+        Timestamp expectedTimestamp = new Timestamp(255073580786543l);
+        Timestamp actualTimestamp = (Timestamp) session.getAttribute("timestamp");
+        assertEquals(expectedTimestamp.getTime(), actualTimestamp.getTime());
+
+        Time expectedTime = new Time(345033920786213l);
+        Time actualTime = (Time) session.getAttribute("time");
+        assertEquals(expectedTime.getTime(), actualTime.getTime());
+
+        BigInteger expectedBigint = new BigInteger("1234567891011121314151617181920");
```

Comment:
Probably should have a negative BigInteger and BigDecimal test to make sure it works correctly.